src/desktop/space/mod.rs

```diff
@@ -385,38 +400,24 @@ impl<E: IsAlive> IsAlive for InnerElement<E> {
     }
 }
 
-impl<E: SpaceElement> SpaceElement for InnerElement<E> {
+impl<E: SpaceElement> InnerElement<E> {
+    // the inner geometry of the element in space coordinates
     fn geometry(&self) -> Rectangle<i32, Logical> {
         let mut geo = self.element.geometry();
-        geo.loc += self.location;
+        geo.loc = self.location;
+        geo.size += geo.loc.to_size();
```

Comment:
This will increase the size of the geometry by the element location, I don't think that is what we want.

Beside that I don't get why this is needed or what it tries to solve. For `weston-terminal` the `element.geometry()`  will be (32,32)x(742,517). If we place the window at (0,0) it will be renderer at (-32,32) with a size of (806,581), which seems correct. For calculating the occupied region in the space, ignoring anything outside of the window geometry, we would use the geometry as defined by the xdg protocol. If we want to have that space relative we should end up with (0,0)x(742,517). There is no need to extend the size by anything imo, everything outside of the original size should not be considered for layouting. So the visible region of the window in a space would be (self.location)x(self.element.geometry.size())

Reply:
Yeah right. I was for some reason expecting the geometry to be represented by two points instead of a point and a size. Of course the size doesn't change by translating. right.

Reply:
fixed